pkg/serviceactivities/shutdown/shutdown.go

```diff
@@ -38,18 +40,24 @@ func New() *ServiceActivity {
 
 // Run runs the shutdown service activity
 func (s *ServiceActivity) Run(ctx context.Context) error {
-	// listen for interrupts and gracefully shutdown server
-	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt, syscall.SIGTERM, syscall.SIGHUP)
+	// listen for interrupt, terminated, and hangup signals and gracefully shutdown server
```

Comment:
Can't you cant the mono channel and check what the type of the signal is? It seems better than three signal.Notify calls+channels.

Reply:
Different os implementations handle what it sends back differently, unfortunately.  It didn't look like something I could trust. :(

Reply:
Really? I'd expect if we pass `os.Interrupt`, `syscall.SIGTERM`, and `syscall.SIGHUP` we'd only every get those values... maybe I'm wrong though?